### PR TITLE
Add sac email changes to SendNotifications

### LINKED ISSUE #7: E-Mail Update: 2 SAC E-Mails
New SAC sponsorship e-mail with updated text (text found [here](https://docs.google.com/document/d/1NgA1IQB6lKGmYcaOxPu7Jaff9W8YOscOUuMz2S8BMk4/edit
)), which has an attached 306 ([OF0306.pdf](https://github.com/GSA/Suitability/files/1170462/OF0306.pdf)), will be sent to the following applicant types: 
     Regions 1-10 (Zones A, B, C, D) NOT in Major Org P - NEW 
     Regions NCR & CO (Zone E) - Everybody gets the NEW 

Old SAC e-mails (those currently being used) will be sent to the remaining applicant types: 
     Regions 1-10 (Zones A, B, C, D) in Major Org P - OLD

### PR DESCRIPTION
**Pull request by jgoodell-gsa**
Closes #7

### CODE REVIEW

**Comment by jgoodell-gsa** [comment]
Ready to merge

File: Suitability/SendNotification.cs [hunk 1]
```diff
@@ -289,11 +289,57 @@ public void SendSponsorshipNotification()
...
+                                body = File.ReadAllText(onboardingLocation + @"\SAC.html");
+
+                                ////Has a specific subject
+                                subject = "[Name (first, middle, last, suffix)] - GSA Special Agreement Check (SAC) Fitness Determination Applicant Instructions (less than 6 month)";
+                            }
+                            break;
+                        case "CO":
+                        case "NCR":
+                            //If Region is "CO" or "NCR" then send new sac email and attachment regardless of MajorOrg
+                            {
```
Comment (tsaunders-gsa) [inline Suitability/SendNotification.cs]: Please remove this bracket.

File: Suitability/SendNotification.cs [hunk 2]
```diff
@@ -289,11 +289,57 @@ public void SendSponsorshipNotification()
...
+                                //use SAC_PBS email
+                                body = File.ReadAllText(onboardingLocation + @"\SAC_PBS.html");
+                                //include OF0306.pdf attachment
+                                emailAttachments.Append(onboardingLocation + @"\OF0306.pdf");
+
+                                ////Has a specific subject
+                                subject = "[Name (first, middle, last, suffix)] - GSA Special Agreement Check (SAC) Fitness Determination Applicant Instructions (less than 6 month)";
+                            }
```
Comment (tsaunders-gsa) [inline Suitability/SendNotification.cs]: Please remove this bracket.

File: Suitability/SendNotification.cs [hunk 3]
```diff
@@ -289,11 +289,57 @@ public void SendSponsorshipNotification()
...
+                            if (!personInfo.MajorOrg.ToLower().Equals("p"))
+                            {
+                                //use SAC_PBS email
+                                body = File.ReadAllText(onboardingLocation + @"\SAC_PBS.html");
+
+                                //include OF0306.pdf attachment
+                                emailAttachments.Append(onboardingLocation + @"\OF0306.pdf");
+
+                                ////Has a specific subject
+                                subject = "[Name (first, middle, last, suffix)] - GSA Special Agreement Check (SAC) Fitness Determination Applicant Instructions (less than 6 month)";
```
Comment (tsaunders-gsa) [inline Suitability/SendNotification.cs]: Subject is the same please move outside of the If statement.

File: Suitability/SendNotification.cs [hunk 4]
```diff
@@ -289,11 +289,57 @@ public void SendSponsorshipNotification()
...
+                            }
+                            //If Region is 1-10 and MajorOrg is "P" then send old sac email
+                            else
+                            {
+                                //old sac email
+                                body = File.ReadAllText(onboardingLocation + @"\SAC.html");
+
+                                ////Has a specific subject
+                                subject = "[Name (first, middle, last, suffix)] - GSA Special Agreement Check (SAC) Fitness Determination Applicant Instructions (less than 6 month)";
```
Comment (tsaunders-gsa) [inline Suitability/SendNotification.cs]: Subject is the same please move outside of the else statement.

File: Suitability/SendNotification.cs [hunk 5]
```diff
@@ -307,39 +307,31 @@ public void SendSponsorshipNotification()
                             {
```
Comment (tsaunders-gsa) [inline Suitability/SendNotification.cs]: Please convert the switch to use lambda expresions

Example
` List<PersonDetails> persInfo = new List<PersonDetails>();

                   if (persInfo.Where(e => new[] { "01", "02", "03" }.Contains(e.Region) && !e.MajorOrg.ToLower().Equals('p')).ToList().Count > 1)`

**Comment by jgoodell-gsa** [comment]
Ready for review

File: Suitability/PersonDetails.cs [hunk 6]
```diff
@@ -27,7 +27,7 @@ public static PersonDetails Adjudication(IDataRecord record)
                 SubjectName = record["Subject Name"].ToString(),
                 Position = record["Position"].ToString(),
                 HomeEMail = record["Home E-Mail"].ToString(),
-                IsCitizen = (bool)record["Is Citizen"],
+                IsCitizen = (sbyte)record["Is Citizen"] == 1,
```
Comment (tsaunders-gsa) [inline Suitability/PersonDetails.cs]: This needs to be set back to bool and cannot be set to 1.  It has to be set to whatever is in the database.

File: Suitability/PersonDetails.cs [hunk 7]
```diff
@@ -40,20 +40,28 @@ public static PersonDetails Adjudication(IDataRecord record)
...
-                Region = record["Region"].ToString(),
-                MajorOrg = record["Major Org"].ToString()
-            };
+                return new PersonDetails
+                {
+                    FullName = record["Full Name"].ToString(),
+                    SubjectName = record["Subject Name"].ToString(),
+                    Position = record["Position"].ToString(),
+                    HomeEMail = record["Home E-Mail"].ToString(),
+                    IsCitizen = (sbyte)record["Is Citizen"] == 1,
```
Comment (tsaunders-gsa) [inline Suitability/PersonDetails.cs]: This needs to be set back to bool and cannot be set to 1.  It has to be set to whatever is in the database.

File: Suitability/SendNotification.cs [hunk 8]
```diff
@@ -289,9 +290,25 @@ public void SendSponsorshipNotification()
                     break;
                 case "sac":
                 case "fingerprint":
-                    body = File.ReadAllText(onboardingLocation + @"\SAC.html");
 
-                    //Has a specific subject
+                    List<PersonDetails> pd = new List<PersonDetails>() { personInfo };
+                    if (pd.Where(e => new[] { "01", "02", "03", "04", "05", "06", "07", "08", "09", "10" }.Contains(e.Region) && e.MajorOrg.ToLower() != "p").ToList().Count > 0)
```
Comment (tsaunders-gsa) [inline Suitability/SendNotification.cs]: Why wasn't NCR and CO included in here?  This way you do not have to do another where clause.

File: Suitability/SendNotification.cs [hunk 9]
```diff
@@ -289,9 +290,25 @@ public void SendSponsorshipNotification()
...
+                    {
+                        UseSacPBSEmail(emailAttachments, ref body);
+                    }
+                    else
+                    {
+                        if (pd.Where(e => new[] { "CO", "NCR" }.Contains(e.Region.ToUpper())).ToList().Count > 0)
```
Comment (tsaunders-gsa) [inline Suitability/SendNotification.cs]: This should be moved up to the other where clause.  This else should only call UseSacEmail(ref body);

**Pull request merged**